src/wmtk/io/EdgeMeshReader.cpp

```diff
@@ -0,0 +1,86 @@
+#include "EdgeMeshReader.hpp"
+#include <spdlog/spdlog.h>
+
+namespace wmtk {
+EdgeMeshReader::EdgeMeshReader(const std::string& filename, const file_type type)
+    : m_filename(filename)
+    , m_type(type)
+{}
+void EdgeMeshReader::read(Eigen::Matrix<long, -1, -1>& E, Eigen::MatrixXd& V)
+{
+    std::vector<std::pair<long, long>> edges;
+    std::vector<std::vector<double>> vertices;
+    char* line_buffer = nullptr;
+    std::fstream f(m_filename);
+    if (!f) {
+        throw std::runtime_error("can't open file!");
+    }
+    if (m_type == OBJ) {
+        std::string buffer;
+        while (!f.eof()) {
+            getline(f, buffer);
+            // here is for debugging
+            // spdlog::info("lineinfo: {}", buffer[0]);
+            if (buffer[0] != 'v' && buffer[0] != 'l') continue;
+            // read obj
+            line_buffer = new char[buffer.size()];
+            strcpy(line_buffer, buffer.c_str());
```

Comment:
You can use a `std::string` just like an array. There is no need to copy the string into a char array.
For example, you can also just write `buffer[0] == 'v'`.